### PR TITLE
♻️ Change Bank Account Request to Bank Account Connection Intent

### PR DESCRIPTION
**Pull request by MeganSteenkamp**
Refactor Bank Account Request to Bank Account Connection Intent.
See Notion API documentation for API design changes.

### CODE REVIEW

File: src/api/bank-accounts/bank-account-connection-intents.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+| type             | String             | The type of Bank Account Connection Intent. See [Bank Account Connection Intent Types][].                            |
+| status           | String             | The current status of the Bank Account Connection Intent. Supported values are `created`, `authorized` and `failed`. |
+| createdAt        | {% dt Timestamp %} | When the Bank Account Connection Intent was created.                                                                 |
+| createdBy        | {% dt CRN %}       | The User or API Key that created the Bank Account Connection Intent.                                                 |
+| updatedAt        | {% dt Timestamp %} | When the Bank Account Connection Intent was updated.                                                                 |
+| updatedBy        | {% dt CRN %}       | The User or API Key that updated the Bank Account Connection Intent.                                                 |
+| authorizationUrl | String             | A URL to the third-party beginning the authorization flow.                                                           |
+
```
Comment (IsaacMurtagh) [inline src/api/bank-accounts/bank-account-connection-intents.md:40]: can we add test as an optionally returned field

File: src/api/bank-accounts/bank-account-connection-intents.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+### Create Bank Account Connection Intent **EXPERIMENTAL**
+
+{% reqspec %}
+  POST '/api/bank-account-connection-intents'
+  auth 'api-key'
+  example {
+    body ({
+      accountId: 'uooxSens6ykJaim1Cu1Q55',
+      type: 'quartz-details',
```
Comment (IsaacMurtagh) [inline src/api/bank-accounts/bank-account-connection-intents.md:58]: quartz-accounts

File: src/api/bank-accounts/bank-account-connection-intents.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+| bankAccountId | String  {% opt %} | The id of the associated [Bank Account][]. Required if type is `quartz-payment`.           |
+| test          | Boolean {% opt %} | A flag which is present if the intention is to create a [Bank Account][] used for testing. |
+
+
+{% h4 Example response payload %}
+
+{% json %}
+  id: 3KVjuKW2CZCJeJVqPxwkX7
+  accountId: B4u4WZCu3joZFVWT3XjWW3
+  type: quartz-details
```
Comment (IsaacMurtagh) [inline src/api/bank-accounts/bank-account-connection-intents.md:79]: quartz-accounts

File: src/api/bank-accounts/bank-account-connection-intents.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+permalink: /api/bank-account-connection-intents
+---
+
+# Bank Account Connection Intents
+{:.no_toc}
+
+A Bank Account Connection Intent facilitates user authorization of access to Bank Accounts.
+Typically the authorization will follow an OAuth flow where the user is
+redirected to a third-party system to authenticate and select one or more bank
+accounts. After a Bank Account Connection Intent has been authorized, one or more
```
Comment (IsaacMurtagh) [inline src/api/bank-accounts/bank-account-connection-intents.md:15]: This is only true for quartz-accounts, can we make more general or add a caveat

Comment (MeganSteenkamp) [reply src/api/bank-accounts/bank-account-connection-intents.md]: Have added that into the section of intent types so it's more specific.

**Review by IsaacMurtagh — APPROVED** [review]

**Pull request merged**